spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerOperations.java

```diff
@@ -124,12 +123,13 @@
 	/**
 	 * Finds all objects of the given type.
 	 * @param entityClass the type of the object to retrieve.
-	 * @param options Cloud Spanner query options with which to conduct the query operation.
+	 * @param options Cloud Spanner query options with which to conduct the query
+	 * operation.
 	 * @param <T> the type of the object to retrieve.
 	 * @return a list of all objects stored of the given type. If there are no objects an
 	 * empty list is returned.
 	 */
-	<T> List<T> queryAll(Class<T> entityClass, SpannerQueryOptions options);
+	<T> List<T> queryAll(Class<T> entityClass, SpannerSortPageQueryOptions options);
```

Comment:
It's interesting that other query methods take `SpanneryQueryOptions` vs this one `SpannerSortPageQueryOptions`. At the least the javadoc should clarify the difference.

Reply:
Yes I will clarify in java doc. The reason is that shorter-named one is all options that don't depend on the SQL string that will be run. The longer-named one includes paging and sorting options, which are only doable in SQL. Any method that takes a SQL string only takes the shorter-named one, which is what we discussed is most intuitive.

Reply:
I think we should be consistent across the query methods. It seems like a soon as I want to provide a query string, I can't take advantage of the paging and sorting options. Why? It seems like a useful functionality that is already there.

Reply:
Because the SQL string already can already specify the sorting and paging. That is the only way to do so for all of spanner client lib.

Reply:
I understand the potential conflict, but don't we already allow it via `@Query` + Pageable in data repositories?

Reply:
Yes, that still works because I moved that paging-sorting string appending/prepending further up in the process so now the final SQL string arrives at the SpannerTemplate not needing the paging/sorting params that the user provided.

Reply:
My point is that one can still do:
```java
@Query("select * from person order by last_name")
List<Person> getPeople(Sort sort);
```

Reply:
Yes, that still works. That was intentionally supported. The SpannerPageableQueryOptions is only used when using SpannerTemplate. I believe it's a different matter.

Reply:
If one is only using `SpannerTemplate` how can they include `SpannerPageableQueryOptions` and provide their own SQL string, without `order by`?

Reply:
They can't. Intentionally they can't. 

All we're doing is appending an "order by" and "limit" to the string anyway. If the user is going to supply the packaged-up sort/pageable options as a java object then he might as well append it to the SQL himself. I generally think it's  good to manipulating a user's given query string as little as possible.